### DISCUSSION TITLE
SSG and fallback render

### DISCUSSION THREAD

**Discussion by armandabric** [vercel/next.js#10684] [Help] [answered]
Hey zeiters :hand:

I'm using the `unstable_getStaticProps` and the `unstable_getStaticPaths` functions in my new project. Theses new APi are really effective and powerful. 

There is one point in the design that still feel uncertain for me: the current implementation of the fallback mechanism. When using the `unstable_getStaticPaths` method, next.js is trying to generate once the page without any query parameters.

As  `unstable_getStaticPaths` is used to pre-generate dynamic paths `/foo/[bar_id]`, rendering a page (executing the page component) without any of the data identifier that qualify that page (`bar_id`) seems counterproductive in the end user point of view. And force us to make or code extremely tolerant or to simply ignore this render: `if (router.isFallback) { return null }` to avoid any risk.

After reading a lot of the past issues and PRs, I discover some of the rationales behind this choice. The main one is that you want to allow a partial static generations of the possible paths. That a really good ideal. And will make this new feature really powerful.  

Another way to allow it could be:
- to serve the page as a lambda that will use the `unstable_getStaticProps` API and render like usual the component
- to return automatically a configurable placeholder component and render the page on client side
- add a way to optout/optin this fallback parameter (when I'm rending all the page of the site I did not need it)

Can you give us more details on where you want to go with the fallback feature? 
 

Thanks for your amazing work on Next.js!

**Comment by Timer** [comment, marked as answer]
We're about to land support for disabling the fallback behavior on sites that don't need it:
https://github.com/zeit/next.js/pull/10701

**Reply by armandabric** [reply]
I see that I'm just too on the edge for this feature! :smiley: Thaks again for your amazing work
